typesystem/fields.py

```diff
@@ -655,16 +655,28 @@ def __init__(self, **kwargs: typing.Any) -> None:
 class Nested(Field):
     errors = {"null": "May not be null."}
 
-    def __init__(self, schema: typing.Any, **kwargs: typing.Any) -> None:
+    def __init__(
+        self, schema: typing.Any, namespace: typing.Mapping = None, **kwargs: typing.Any
+    ) -> None:
         super().__init__(**kwargs)
         self.schema = schema
+        self.namespace = namespace
 
     def validate(self, value: typing.Any, *, strict: bool = False) -> typing.Any:
+        if isinstance(self.schema, str):
+            assert (
+                self.namespace is not None
+            ), "String-references can only be used within a schema namespace."
+            schema = self.namespace[self.schema]
```

Comment:
you may want to consider to assign to `self.schema` as well to only do this one to ‘settle’ the names on first use.

Reply:
Yes probably a good plan. Less scope for behavioural oddities then.